apps/storefront/src/pages/ShoppingListDetails/components/ReAddToCart.tsx

```diff
@@ -211,6 +213,29 @@ export default function ReAddToCart(props: ShoppingProductsProps) {
     setValidateFailureProducts(newProduct);
   };
 
+  const shouldRedirectToCheckout = () => {
+    handleCancelClicked();
+    if (
+      allowJuniorPlaceOrder &&
+      submitShoppingListPermission &&
+      shoppingListInfo?.status === ShoppingListStatus.Approved
+    ) {
+      window.location.href = CHECKOUT_URL;
+    } else {
+      snackbar.success(b3Lang('shoppingList.reAddToCart.productsAdded'), {
+        action: {
+          label: b3Lang('shoppingList.reAddToCart.viewCart'),
+          onClick: () => {
+            if (window.b2b.callbacks.dispatchEvent('on-click-cart-button')) {
+              window.location.href = CART_URL;
+            }
+          },
+        },
+      });
+      b3TriggerCartNumber();
+    }
+  };
+
   const handRightClick = async () => {
```

Comment:
Can we rename to `handlePrimaryAction` to be more descriptive.
Let's try to do this for all the usage of `B3Dialog` so we can eventually rename the component's prop as well.

```suggestion
  const handlePrimaryAction = async () => {
```